src/block/block_ext.c

```diff
@@ -413,6 +413,12 @@ __wt_block_off_remove_overlap(
     /* If "before" or "after" overlaps, retrieve the overlapping entry. */
     if (before != NULL && before->off + before->size > off) {
         WT_RET(__block_off_remove(session, block, el, before->off, &ext));
+        if (ext->off + ext->size < off + size) {
```

Comment:
To avoid the cost of performing this check in a release build, I would suggest using WT_ASSERT here instead. The below `>` change will already ensure no negative sized extents are placed into the list and an assert will immediatly panic here in our testing so it would be easy to debug. Also please add a  comment explaining this problem case.

Reply:
hi, @Sean04 
1. Ultimately, it will be executed "abort()" function, WT_BLOCK_RET can obtain WT_ ASSERT has the same effect and can provide the desired log information.
2. Maintain consistent code style. for example, https://github.com/wiredtiger/wiredtiger/blob/develop/src/block/block_ext.c#L1036-L1050

So I think using WT_BLOCK_RET may be better, What's your opinion？
thanks.

Reply:
The difference would be that WT_ASSERT would only run when WiredTiger is compiled with the HAVE_DIAGNOSITC flag. Do you think this is a necessary check to have in release builds?

Reply:
hi, @Sean04 
Thank you for your reply. I think that we should add this judgment even when it not in HAVE_DIAGNOSITC flag，it is an aberration.

thanks.

Reply:
Thanks for point out the check in [block_ext.c](https://github.com/wiredtiger/wiredtiger/pull/9898#:~:text=https%3A//github.com/wiredtiger/wiredtiger/blob/develop/src/block/block_ext.c%23L1036%2DL1050), however, unless the check here is necessarily avoiding a corruption we should use an assert. A corruption here would indicate a bug in the block manager that should be covered by our testing, therefore we should leave it as a debugging check.